Sources/VariadicsGenerator/VariadicsGenerator.swift

```diff
@@ -360,21 +352,23 @@ struct VariadicsGenerator: ParsableCommand {
     let matchType = arity == 0 ? baseMatchTypeName : "(\(baseMatchTypeName), \(quantifiedCaptures))"
     output("""
       \(arity == 0 ? "@_disfavoredOverload" : "")
-      public func \(kind.rawValue)<\(genericParameters(withConstraints: true))>(
-        _ component: Component
+      public func \(kind.rawValue)<\(genericParams)>(
+        _ component: Component,
+        _ kind: QuantificationKind = .eager
```

Comment:
Should `kind` come before `component`? This would make it consistent with the builder API. For the builder API we are kinda out of options since we'd have to support a trailing closure.

---

So it's

```swift
oneOrMore("x", .eager)
oneOrMore(.eager) { "x" }
```

vs.

```swift
oneOrMore(.eager, "x")
oneOrMore(.eager) { "x" }
```

Reply:
I can't decide about this. On the one hand, it's nice to have them in a consistent order. On the other, I think it reads better to have the thing you're matching come first, and only shift it to the end when it's in the trailing closure position. IIUC, this is how SwiftUI handles it pretty much across the board — for example, when creating a `Button` with just a label, the string is the first parameter, otherwise, the builder closure comes at the end.

Let's punt on this for now to see how many different calls like this we have.